src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -1,21 +1,355 @@
 package storage_mining
 
-import base_mining "github.com/filecoin-project/specs/systems/filecoin_mining"
 import sector "github.com/filecoin-project/specs/systems/filecoin_mining/sector"
-import sealing "github.com/filecoin-project/specs/systems/filecoin_mining/sector"
+import block "github.com/filecoin-project/specs/systems/filecoin_blockchain/struct/block"
+import poster "github.com/filecoin-project/specs/systems/filecoin_mining/storage_proving/poster"
+
+// If a Post is missed (either due to faults being not declared on time or
+// because the miner run out of time, every sector is reported as failing
+// for the current proving period.
+func (sm *StorageMinerActor_I) CronAction() {
+	// TODO: detemine how much pledge collateral to slash
+	// TODO: SendMessage(spa.SlashPledgeCollateral)
+	var powerUpdate uint64 = 0
+
+	for sectorNo, _ := range sm.Sectors() {
+		state := sm.SectorStates()[sectorNo]
+		switch state.StateNumber {
+		case SectorCommittedStateNo:
```

Comment:
this switch is great 👍

Reply:
i think this switch should move into a function `sm.failSectors([]sectorNo)` and this function should just do this:

```go
func (sm *StorageMinerActor_I) CheckPostSubmission() {
  if (post was not submitted) {
    nums := getSectorNums(sm.Sectors())
    sm.failSectors(nums)
  }
}
```

the other case (DeclareFaults) would just do this:

```go
func (sm *StorageMinerActor_I) DeclareFaults(faultSet sector.CompactSectorSet) {
  ...
  sectors := faultSet.SectorsInSet()
  sm.failSectors(sectors)
  ...
}
```